src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,143 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private final float loadFactor;
+    private int threshold;
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.loadFactor = DEFAULT_LOAD_FACTOR;
+        this.threshold = (int) (capacity * loadFactor);
+        this.table = createTable(capacity);
+    }
+
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        this.capacity = initialCapacity;
+        this.loadFactor = loadFactor;
+        this.threshold = (int) (capacity * loadFactor);
+        this.table = createTable(capacity);
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+
+        int hash = hash(key);
+        int index = indexFor(hash, capacity);
 
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (key.equals(node.key)) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+
+        int hash = hash(key);
+        int index = indexFor(hash, capacity);
+
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (key.equals(node.key)) {
+                return node.value;
+            }
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        for (Node<K, V> node = table[0]; node != null; node = node.next) {
+            if (node.key == null) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(0, null, value, 0);
+    }
+
+    private V getForNullKey() {
+        for (Node<K, V> node = table[0]; node != null; node = node.next) {
+            if (node.key == null) {
+                return node.value;
+            }
+        }
+        return null;
```

Comment:
dont' handle null keys separately, you need to use null safe comparison checks like one in Objects.equals so you don't have to treat them separately